hydroflow/src/scheduled/input.rs

```diff
@@ -1,31 +1,54 @@
-use std::{cell::RefCell, rc::Rc};
+use std::{cell::RefCell, marker::PhantomData, rc::Rc, sync::mpsc::SyncSender};
 
 use super::{OpId, Reactor};
 
-pub struct Input<T> {
+pub trait Givable<T> {
```

Comment:
I think `Give` is the more idiomatic name btw

Reply:
Done.